homeassistant/components/wallbox/config_flow.py

```diff
@@ -0,0 +1,114 @@
+"""Config flow for Wallbox integration."""
+import logging
+
+import requests
+import voluptuous as vol
+from wallbox import Wallbox
+
+from homeassistant import config_entries, core, exceptions
+
+from .const import DOMAIN
+
+COMPONENT_DOMAIN = DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        "station": str,
+        "username": str,
+        "password": str,
+    }
+)
+
+
+class PlaceholderHub:
+    """Wallbox Hub class."""
+
+    def __init__(self, station, username, password):
+        """Initialize."""
+        self._station = station
+        self._username = username
+        self._password = password
+
+    def authenticate(self) -> bool:
+        """Authenticate using Wallbox API."""
+        try:
+            wallbox = Wallbox(self._username, self._password)
```

Comment:
Seems like a duplicate of "async_setup_entry" ?

Reply:
not completely, but to some extend I guess. This is mainly based on the scaffolding component for the config flow. Here I try to validate input in the config flow -> prevent that the user enters incorrect data and ends up with a non-working integration. To my understanding the config flow part is separate from the init file, so there is some duplicate work here.

Reply:
Yes your understanding is correct..the init file is used when your integration start up, and config_flow when a user want to configure. I guess I did not write clear enough, duplicate code is more likely to grow different over time, so it is better to have a common function.